cfe_internal/core/watchdog/watchdog.cf

```diff
@@ -17,18 +17,99 @@ bundle agent cfe_internal_core_watchdog(state)
       "use_cfe_internal_core_watchdog_cron_d"
         expression => "have_cron_d._stdlib_has_path_pgrep";
 
+      # We use the aix specific watchdog implementation when it's aix and we are
+      # not using the cron.d implementation.
+
+      "use_cfe_internal_core_watchdog_aix"
+        expression => "!use_cfe_internal_core_watchdog_cron_d.aix";
+
   methods:
     use_cfe_internal_core_watchdog_cron_d::
       "any" usebundle => cfe_internal_core_watchdog_cron_d( $(state) );
 
+    use_cfe_internal_core_watchdog_aix::
+      "any" usebundle => cfe_internal_core_watchdog_aix( $(state) );
+
   reports:
     DEBUG|DEBUG_cfe_internal_core_watchdog::
       "DEBUG $(this.bundle): Watchdog '$(state)'";
       "DEBUG $(this.bundle): Invalid state '$(state)' only enabled|disabled allowed"
         ifvarclass => "invalid_state";
 
-    !use_cfe_internal_core_watchdog_cron_d::
-      "WARNING $(this.bundle): Currently only supports /etc/cron.d on systems that have pgrep in the the stdlib paths bundle.";
+    !(use_cfe_internal_core_watchdog_cron_d|aix)::
+      "WARNING $(this.bundle): Currently only supports /etc/cron.d on systems that have pgrep in the the stdlib paths bundle and AIX hosts.";
+}
+
+bundle agent cfe_internal_core_watchdog_aix(state)
+# @brief Manage watchdog state on aix
+# @param state enabled|disabled
+# - When enabled a cron job will be present to start cf-execd if it's not running.
+# - When disabled cron jobs including the the text CFEngine watchdog will not be present.
```

Comment:
Double *the*

Reply:
Something more like "When disabled cron jobs including the CFEngine watchdog will not be present."

Reply:
Thank you.